src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -296,4 +316,49 @@ trait ResourceRoutes extends UserInfoDirectives with SecurityDirectives with Sam
         })
       }
     }
+
+  def getResourceParent(resource: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route =
+    get {
+      requireAction(resource, SamResourceActions.getParent, userInfo.userId, samRequestContext) {
+        complete(resourceService.getResourceParent(resource, samRequestContext).map {
+          case Some(response) => StatusCodes.OK -> response
+          case None => throw new WorkbenchExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "resource parent not found"))
+        })
+      }
+    }
+
+  def setResourceParent(resource: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route =
+    put {
+      entity(as[FullyQualifiedResourceId]) { newResourceParent =>
+        withCurrentParentOption(resource, samRequestContext) {
+          case Some(currentResourceParent) =>
+            requireAction(currentResourceParent, SamResourceActions.removeChild, userInfo.userId, samRequestContext) {
+              setResourceParentInternal(resource, newResourceParent, userInfo, samRequestContext)
+            }
+          case None => setResourceParentInternal(resource, newResourceParent, userInfo, samRequestContext)
+        }
+      }
+    }
+
+  private def setResourceParentInternal(resource: FullyQualifiedResourceId, newResourceParent: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route = {
+    requireAction(newResourceParent, SamResourceActions.addChild, userInfo.userId, samRequestContext) {
+      requireAction(resource, SamResourceActions.setParent, userInfo.userId, samRequestContext) {
+        complete(resourceService.setResourceParent(resource, newResourceParent, samRequestContext).map(_ => StatusCodes.NoContent))
+      }
+    }
+  }
+
+  def deleteResourceParent(resource: FullyQualifiedResourceId, userInfo: UserInfo, samRequestContext: SamRequestContext): server.Route =
```

Comment:
This just removes relationship between child and parent, it doesn't remove either resource from the database